bas/Model/Recorders.bas

```diff
@@ -441,250 +441,863 @@ deformations
+*if(GenData(Stresses_in_layer_A,int)==1||GenData(Stresses_in_layer_B,int)==1||GenData(Stresses_in_layer_C,int)==1||GenData(Strains_in_layer_A,int)==1||GenData(Strains_in_layer_B,int)==1||GenData(Strains_in_layer_C,int)==1||GenData(Temp_Elong_and_KtKc_in_layer_A,int)==1||GenData(Temp_Elong_and_KtKc_in_layer_B,int)==1||GenData(Temp_Elong_and_KtKc_in_layer_C,int)==1)
+for {set gaussID 1} {$gaussID<=4} {incr gaussID} {
+*loop elems
+*if(strcmp(ElemsMatProp(Element_type:),"Shell")==0)
+*set var SelectedSection=tcl(FindMaterialNumber *ElemsMatProp(Type) *DomainNum)
+*loop materials *NotUsed
+*set var SectionID=tcl(FindMaterialNumber *MatProp(0) *DomainNum)
+*if(SelectedSection==SectionID)
+*if(MatProp(Decking_thickness,Real) > 0)
+*set var Decklayers = 1
+*else
+*set var Decklayers = 0
+*endif
+*if(MatProp(Bot_cover_layers,int) > 0)
+*set var Botcoverlayers = MatProp(Bot_cover_layers,int)
+*else
+*set var Botcoverlayers = 0
+*endif
+*if(MatProp(Bot_transverse_bar_diameter,Real) > 0)
+*set var Bottranssteellayers = 1
+*else
+*set var Bottranssteellayers = 0
+*endif
+*if(MatProp(Bot_longitudinal_bar_diameter,Real) > 0)
+*set var Botlongsteellayers = 1
+*else
+*set var Botlongsteellayers = 0
+*endif
+*if(MatProp(Core_layers,int) > 0)
+*set var corelayers = MatProp(Core_layers,int)
+*else
+*set var corelayers = 0
+*endif
+*if(MatProp(Top_longitudinal_bar_diameter,Real) > 0)
+*set var Toplongsteellayers = 1
+*else
+*set var Toplongsteellayers = 0
+*endif
+*if(MatProp(Top_transverse_bar_diameter,Real) > 0)
+*set var Toptranssteellayers = 1
+*else
+*set var Toptranssteellayers = 0
+*endif
+*if(MatProp(Top_cover_layers,int) > 0)
+*set var Topcoverlayers = MatProp(Top_cover_layers,int)
+*else
+*set var Topcoverlayers = 0
+*endif
+*break
+*endif
+*end materials
+*break
+*endif
+*end elems
+*endif
+*#-------------------------------------------------------------------------------------------------------------------------
 *if(GenData(Stresses_in_layer_A,int)==1)
-
-*set var layer = GenData(Stress_layer_A,int)
-*for(i=1;i<=4;i=i+1)
-recorder Element -file ShellMITC4_stress_Layer*layer_GP*i.out -time -ele *\
+*set var loopnum = 0
+recorder Element -file *\
 *loop elems
 *if(strcmp(ElemsMatProp(Element_type:),"Shell")==0)
+*set var loopnum = loopnum + 1
+*if(loopnum == 1)
+*if(strcmp(GenData(Stress_layer_A),"SteelTopLongitudinal")==0)
+*set var layer = Decklayers+Botcoverlayers+Bottranssteellayers+Botlongsteellayers+corelayers+Toplongsteellayers
+ShellMITC4_steel_stress_Layer*layer_GP$gaussID.out -time -ele *\
+*elseif(strcmp(GenData(Stress_layer_A),"SteelTopTransverse")==0)
+*set var layer = Decklayers+Botcoverlayers+Bottranssteellayers+Botlongsteellayers+corelayers+Toplongsteellayers+Toptranssteellayers
+ShellMITC4_steel_stress_Layer*layer_GP$gaussID.out -time -ele *\
+*elseif(strcmp(GenData(Stress_layer_A),"SteelBotLongitudinal")==0)
+*set var layer = Decklayers+Botcoverlayers+Bottranssteellayers+Botlongsteellayers
```

Comment:
This is incorrect. Remember layers are from bottom to top: bottom cover, bottom longitudinal, bottom transverse, core, top longitudinal, top transverse, top cover
So:
*elseif(strcmp(GenData(Stress_layer_A),"SteelBotLongitudinal")==0)
*set var layer = Decklayers+Botcoverlayers+Bottranssteellayers+Botlongsteellayers

should be:
*elseif(strcmp(GenData(Stress_layer_A),"SteelBotLongitudinal")==0)
*set var layer = Decklayers+Botcoverlayers+Botlongsteellayers

Reply:
Ok, I will correct it! I got confused with the pattern of steel layers arrangement